src/int_allocator.rs

```diff
@@ -1,14 +1,21 @@
-use std::vec;
-
 use crate::allocator::{Allocator, SExp};
 
+#[derive(Clone, Copy)]
+pub struct IntAtomBuf {
```

Comment:
You might consider using tuple-type structs here like
```
pub struct IntAtomBuf (u32, u32);
```
Get the values out with `int_atom_buf.0` and `int_atom_buf.1`.

Reply:
yeah, I hadn't realized you could do that at the time. But I also like having the names. I think there's a risk the members could, incorrectly, be interpreted as `start` and `length` for instance.